static/gsAdmin/components/users/userEmails.spec.tsx

```diff
@@ -0,0 +1,109 @@
+import {UserFixture} from 'sentry-fixture/user';
+
+import {
+  render,
+  renderGlobalModal,
+  screen,
+  userEvent,
+  waitFor,
+} from 'sentry-test/reactTestingLibrary';
+
+import ModalStore from 'sentry/stores/modalStore';
+
+import UserEmails from 'admin/components/users/userEmails';
+
+describe('UserEmails', function () {
+  const mockUser = UserFixture({
+    id: '1',
+    email: 'primary@example.com',
+    emails: [
+      {id: '1', email: 'primary@example.com', is_verified: true},
+      {id: '2', email: 'secondary@example.com', is_verified: false},
+    ],
+  });
+
+  const mockPanel = ({children}: {children: React.ReactNode}) => <div>{children}</div>;
+
+  beforeEach(() => {
+    ModalStore.reset();
+    MockApiClient.clearMockResponses();
+  });
+
+  it('renders email addresses correctly', function () {
+    render(<UserEmails Panel={mockPanel} user={mockUser} />);
+
+    expect(screen.getByText('primary@example.com')).toBeInTheDocument();
+    expect(screen.getByText('secondary@example.com')).toBeInTheDocument();
+    expect(screen.getByText('— primary')).toBeInTheDocument();
```

Comment:
you can use [textWithMarkupMatcher](https://github.com/getsentry/sentry/blob/b8aec26acf1241d58e7cba526a18dc820339410d/tests/js/sentry-test/utils.tsx#L27) to match text within multiple nodes, to make sure that the correct email is being labeled as primary